app/src/main/java/com/mtesitoo/fragment/ProductFragment.java

```diff
@@ -115,4 +126,24 @@ public void onError(Exception e) {
         mProductListAdapter = new ProductListAdapter(getActivity(), new ArrayList<Product>());
         setListAdapter(mProductListAdapter);
     }
+
+    private BroadcastReceiver mMessageReceiver = new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            // Get extra data included in the Intent
+            String message = intent.getStringExtra("result");
+            Log.d("LocalBroadcastManager", "result : " + message);
+            reloadProductList();
+        }
+    };
+
+    private void reloadProductList() {
+        if (pd != null) {
+            pd.cancel();

```

Comment:
wondering if calling `cancel()` on `ProgressDialog` makes it `null` ? If not, then we may want to create a new instance of `ProgressDialog` only when its `null`. A check for `null` before creating an instance for ProgressDialog may make sense in that case.

Reply:
Hmm.. I think what I should do is check if it is == null and create an instance and do the rest of the actions outside the if block. Also, in my next pull request for another issue ProgressDialog will be removed anyways.